ic-asset/src/asset_config.rs

```diff
@@ -227,6 +232,10 @@ impl AssetConfig {
             (_, Maybe::Null) => self.headers = None,
             (_, Maybe::Absent) => (),
         };
+
+        if self.ignore != Some(true) && other.ignore.is_some() {
+            self.ignore = other.ignore;
+        }
```

Comment:
Because it shouldn't be possible to unignore previously ignored files. 
I tried to follow the logic from https://git-scm.com/docs/gitignore
> An optional prefix "!" which negates the pattern; any matching file excluded by a previous pattern will become included again. It is not possible to re-include a file if a parent directory of that file is excluded. Git doesn’t list excluded directories for performance reasons, so any patterns on contained files have no effect, no matter where they are defined. Put a backslash ("\") in front of the first "!" for patterns that begin with a literal "!", for example, "\!important!.txt".

accepting proposed change would fail the following test 
https://github.com/dfinity/agent-rs/blob/SDK-592-configurable-hidden-directories-files-inclusion-when-uploading-assets/ic-asset/src/sync.rs#L551
but changing 
```rust
let mut expected_asset_descriptors = vec![
    AssetDescriptor::default_from_path(&assets_dir, "file"),
    AssetDescriptor::default_from_path(&assets_dir, ".hidden-dir-flat/.hfile"),
];
``` 
to 
```rust
let mut expected_asset_descriptors = vec![
    AssetDescriptor::default_from_path(&assets_dir, "file"),
    AssetDescriptor::default_from_path(&assets_dir, ".hidden-dir-flat/.hfile"),
    AssetDescriptor::default_from_path(&assets_dir, ".hidden-dir-flat/file"),
];
```
would make it pass again.

Let me know how you'd want to proceed here.

Reply:
This is the exact case that I was wondering about.

I would expect the following:
1. If a directory is ignored, file and directories within it cannot be un-ignored
2. Properties from a rule matching a file in a subdirectory override properties from a rule matching a file in a parent directory

The proposed change achieves this, and is also consistent with the referenced rules for .gitnore:

> ... any matching file excluded by a previous pattern will become included again. It is not possible to re-include a file if a parent directory of that file is excluded.

What do you think?

Reply:
sold!

Reply:
done in https://github.com/dfinity/agent-rs/pull/361/commits/1c95f701ace6d9ed9c81fe810fbc60c814250f89